larq/quantizers.py

```diff
@@ -156,10 +152,14 @@ def ste_sign(x, clip_value=1.0):
       Activations Constrained to +1 or -1](http://arxiv.org/abs/1602.02830)
     """
 
-    if clip_value is not None:
-        x = tf.clip_by_value(x, -clip_value, clip_value)
+    @tf.custom_gradient
+    def _call(x):
+        def grad(dy):
+            return _clipped_gradient(x, dy, clip_value)
 
-    return _binarize_with_identity_grad(x)
+        return math.sign(x), grad
+
+    return _call(x)
```

Comment:
Is there a reason that `ste_sign` is implemented as `return _call(x)` where `_call` gets the `tf.custom_gradient` decorator, whereas `approx_sign` below gets the decorator directly on it instead of this subfunction?

Reply:
Unfortunately `tf.custom_gradient` only supports key word arguments in eager mode. So we need to wrap it into the `_call` function to make it work. Checkout fa5ef93566fefecef941d33739876243527a4b79